### DISCUSSION TITLE
Elevate permissions

### DISCUSSION THREAD

**Discussion by Sammy990** [wlcrs/huawei_solar#328] [Setting up the integration]
Is it possible to enable the elevation after the integration has already been set up without?

I have it running for several weeks and don't want to lose the data. So I was looking for a ways to elevate the additional permissions without completely deleting the integration and setting up from scratch.

Thanks

**Comment by Sammy990** [comment]
@wlcrs ?

**Comment by janrosseel** [comment]
I have the same question. I have a setup that is running fine for 3 months in "read-only" mode. Didn't select elevate permissions" at install specifically to keep things in read-only as I had an issue with the inverter underperforming, and didn't want to interfere with the config. 

But now the inverter is fine (and it was actually this integration that provided much more detail than the FusionSolar app that allowed to pinpoint the problem), so I'd like to optimize the battery charging from within HA. So I need access to these configuration panels listed in the documentation, but they're nowhere to be seen. 

How can we change this after initial install?

**Comment by netmb** [comment]
I've done that many times: Simply remove the inverter from the integration and add it back with elevated permissions. Entity-IDs remain the same.

**Comment by janrosseel** [comment]
Thanks for the reply. I'm afraid removing the integration and adding it back in will screw up things, though... 

I have two inverters in the integration. The first one, and then it's (warranty) replacement. So all entity/sensor ID's for the current, working inverter have a _2 suffix. 

I'm afraid that after deleting the integration, and adding it back in, I will now have all entities/sensors with names without the _2 suffix. This will make it difficult to have a continuous history. At least, that's what I'm afraid for. Is this worry justified?

**Comment by henfri** [comment]
Hello,

removing and re-adding (with the checkmark to elevate permissions) did not prompt for a password.
Is there a trick?

Regards,
Hendrik

**Comment by MEKadan** [comment]
> I've done that many times: Simply remove the inverter from the integration and add it back with elevated permissions. Entity-IDs remain the same.


How do you actually do that? Is it the red "Delete" option on the right?

![image](https://github.com/wlcrs/huawei_solar/assets/66167932/41ad05c1-2e71-4355-904e-2c8bacbe35ac)

Edit:
To be on a safe side, it is probably a good idea to do a full backup before attempting to delete anything...

BTW, I vaguely recall that I did not check the "elevated permissions" in the first place, because it would not let me  to proceed from that?
